Src/PerformanceCollector/Unit.Tests/Mocks/PerformanceCollectorMock.cs

```diff
@@ -98,8 +94,97 @@ public IEnumerable<PerformanceCounterData> PerformanceCounters
             }
         }
 
-        public void RefreshPerformanceCounter(PerformanceCounterData pcd, PerformanceCounter pc)
+        public void RefreshPerformanceCounter(PerformanceCounterData pcd)
+        {
+        }
+
+        public void RefreshCounters()
+        {
+        }
+
+        public void LoadDependentInstances()
+        {
+        }
+
+        public void RegisterCounter(
+            string perfCounterName,
+            string reportAs,
+            bool isCustomCounter,
+            out string error)
+        {
+            bool usesInstanceNamePlaceholder;
+            var pc = this.CreateCounter(
+                perfCounterName,
```

Comment:
I am slightly confused with this logic: We create a counter at this place, which is basically kind of validating the performance collector, and returning the coutner. However, later, in RegisterCounter(), we actually create a new performance counter using the category name, and counter name. Please let me know if this is expected, and the reason why we do it twice. Thanks!